### ISSUE TITLE
Android:“physicalViewInsetBottom” calculation logic change cause “fullScreen” Navigation bar area cannot be drawn

### ISSUE THREAD

**Issue by liujiakuoyx** [flutter/flutter#52472]
io.flutter.view.FlutterView:
![image](https://user-images.githubusercontent.com/22370823/76533783-8fa37800-64b3-11ea-9d38-52cf269d41e7.png)

io.flutter.embedding.android.FlutterView:
![image](https://user-images.githubusercontent.com/22370823/76533840-a6e26580-64b3-11ea-9ec4-38be9ce57996.png)

My navigation bar position cannot be drawn when I use the "io.flutter.embedding.android.FlutterView" and using Scaffold widget.
![image](https://user-images.githubusercontent.com/22370823/76535261-9206d180-64b5-11ea-93ce-937a90dba625.png)
The setting "resizeToAvoidBottomInset: false" is effective, but this will cause the soft keyboard to be unable to handle.

I think the value of "viewInsetBottom" should be calculated like in "io.flutter.view.FlutterView" when the navigation bar is hidden.

Flutter 1.12.13+hotfix.8 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 0b8abb4724 (4 weeks ago) • 2020-02-11 11:44:36 -0800
Engine • revision e1e6ced81d
Tools • Dart 2.7.0

Thanks!

**Comment by itsJoKr** [comment]
What's the difference between `io.flutter.view.FlutterView` and `io.flutter.embedding.android.FlutterView` anyway? I couldn't find the answer anywhere

**Comment by liujiakuoyx** [comment]
> What's the difference between `io.flutter.view.FlutterView` and `io.flutter.embedding.android.FlutterView` anyway? I couldn't find the answer anywhere

Please see where the rad arrow points.In `io.flutter.view.FlutterView`, when the navigation bar is hidden, use the "calculateBottomKeyBoardInset" method to determine whether there is currently a soft keyboard，return 0 when the soft keyboard is hidden.
![image](https://user-images.githubusercontent.com/22370823/78148874-ede6c980-7467-11ea-806a-7e0f25112361.png)

**Comment by cetozdemir** [comment]
2 lines code can resolve this problem.

` return Scaffold(
      resizeToAvoidBottomPadding: false, //remove bottom padding
      resizeToAvoidBottomInset: true, // enable keyboard resizing
   );`

**Comment by liujiakuoyx** [comment]
> 2 lines code can resolve this problem.
> 
> ` return Scaffold( resizeToAvoidBottomPadding: false, //remove bottom padding resizeToAvoidBottomInset: true, // enable keyboard resizing );`
This way does not work

**Comment by liujiakuoyx** [comment]
I found a way to work
```
bool canResizeToAvoidBottomInset(BuildContext context) {
  if (context == null) return false;
  var viewInsets = MediaQuery.of(context).viewInsets;
  var insetsBottom = viewInsets.bottom;
  var screenHeight = MediaQuery.of(context).size.height;
  return (screenHeight * 0.18) < insetsBottom;
}

return Scaffold(
     resizeToAvoidBottomInset: canResizeToAvoidBottomInset(context),
     .......
);
```

**Comment by liujiakuoyx** [comment]
fixed by #47683
